networkx/algorithms/tests/test_dag.py

```diff
@@ -655,7 +665,41 @@ def test_is_aperiodic_disconnected2():
     G = nx.DiGraph()
     nx.add_cycle(G, [0, 1, 2])
     G.add_edge(3, 3)
-    assert not nx.is_aperiodic(G)
+    assert nx.is_aperiodic(G)
+
+
+def test_is_aperiodic_acyclic_component():
+    G = nx.DiGraph()
+    G.add_node(0)
+    G.add_edge(1, 1)
+    assert nx.is_aperiodic(G)
+
+
+def test_is_aperiodic_acyclic_component2():
+    G = nx.DiGraph()
+    G.add_node(0)
+    nx.add_cycle(G, [1, 2, 3])
+    nx.add_cycle(G, [2, 3, 4, 5])
+    assert nx.is_aperiodic(G)
+
+
+@pytest.mark.parametrize("seed", range(10))
+def test_is_aperiodic_random(seed):
+    rng = random.Random(seed)
```

Comment:
It's recommended for new code to use [`random.Generator.integers`](https://numpy.org/doc/2.1/reference/random/generated/numpy.random.Generator.integers.html#numpy.random.Generator.integers) instead.

Reply:
I think generators are for numpy random numbers.  This is using Python's random module. And that's fine.

Reply:
Yes, I realized my mistake when seeing the test failures.